subprojects/docs/src/docs/userguide/java.adoc

```diff
@@ -0,0 +1,810 @@
+
+Speaking of `writeTo()`, you can use that to easily write a manifest to disk at any time, like so:
+
+++++
+<sample xmlns:xi="http://www.w3.org/2001/XInclude" id="manifest" dir="userguide/tutorial/manifest" title="Saving a MANIFEST.MF to disk">
+                <sourcefile file="build.gradle" snippet="write"/>
+            </sample>
+++++
+
+[[sec:generating_javadocs]]
+=== Generating API documentation
+
+The Java Plugin provides a `javadoc` task of type api:org.gradle.api.tasks.javadoc.Javadoc[], that will generate standard Javadocs for all your production code, i.e. whatever source is in the `main` source set. The task supports the core Javadoc and standard doclet options described in the http://docs.oracle.com/javase/7/docs/technotes/tools/windows/javadoc.html#options[Javadoc reference documentation]. See api:org.gradle.external.javadoc.CoreJavadocOptions[] and api:org.gradle.external.javadoc.StandardJavadocDocletOptions[] for a complete list of those options.TODO:Not sure if the user can rely on `options` being of type `StandardJavadocDocletOptions` rather than `MinimalJavadocOptions`
+
+As an example of what you can do, imagine you want to use Asciidoc syntax in your Javadoc comments. To do this, you need to add Asciidoclet to Javadoc's doclet path. Here's an example that does just that:
+
+++++
+<sample id="usingCustomDoclet" dir="userguide/java/basic" title="Using a custom doclet with Javadoc">
+    <sourcefile file="build.gradle" snippet="using-custom-doclet"/>
+</sample>
+++++
+
+You don't have to create a configuration for this, but it's an elegant way to handle dependencies that are required for a unique purpose.
+
+You might also want to create your own Javadoc tasks, for example to generate API docs for the tests:
+
+++++
+<sample id="customJavadocTask" dir="userguide/java/basic" title="Defining a custom Javadoc task">
+    <sourcefile file="build.gradle" snippet="defining-custom-javadoc-task"/>
+</sample>
+++++
+
+These are just two non-trivial but common customizations that you might come across. 
+
+[[sec:cleaning_java_build]]
+=== Cleaning the build
+
+The Java Plugin adds a `clean` task to your project by virtue of applying the Base Plugin. This task simply deletes everything in the `$buildDir` directory, hence why you should always put files generated by the build in there. The task is an instance of api:org.gradle.api.tasks.Delete[] and you can change what directory it deletes by setting its `dir` property.
+
+[[sec:building_java_libraries]]
```

Comment:
We may want to split out the following section as separate chapters.

Reply:
I'd rather keep it here. The sections aren't long and are an important topic for first look readers. These sections just don't seem to warrant a chapter to themselves, except perhaps individually, but we have the plugin chapters for that.

Reply:
We can split things up in a different PR if needed. It's fine to keep it as-is for now.